### PR TITLE
Whitelist System

### PR DESCRIPTION
**Pull request by robbybaseplate**

### CODE REVIEW

**Comment by Blumlaut** [comment]
looks good to me, will test and merge if everything checks out! i have also linked issue #438 so once this is merged, the issue will automatically solve as well.

File: server/admin_server.lua [hunk 1]
```diff
@@ -920,7 +920,17 @@ Citizen.CreateThread(function()
 			end
 		end
 		
-		deferrals.done()
+		if GetConvar("ea_whitelist", "false") == "true" then
```
Comment (Blumlaut) [inline server/admin_server.lua:923]: i'd probably call this `ea_enableWhitelist` though, to keep in-line with existing convars

File: server/admin_server.lua [hunk 2]
```diff
@@ -920,7 +920,17 @@ Citizen.CreateThread(function()
...
+			if DoesPlayerHavePermission(player, "player.whitelist") then
```
Comment (Blumlaut) [inline server/admin_server.lua:924]: i'd probably also rename this to player.allowlist to be in-line with what FiveM calls whitelists

File: server/admin_server.lua [hunk 3]
```diff
@@ -920,7 +920,17 @@ Citizen.CreateThread(function()
...
+				deferrals.done()
+			else
+				deferrals.done(GetLocalisedText("whitelist"))
```
Comment (Blumlaut) [inline server/admin_server.lua:927]: locale would have to be changed to `allowlist` as well

**Review by Blumlaut — APPROVED** [review]
apart from these quirks, lgtm, if you dont want to make the edit i'll merge & change it myself.

**Comment by robbybaseplate** [comment]
Changed and tested

**Pull request merged**